apps/client/src/common/utils/aliases.ts

```diff
@@ -26,4 +29,42 @@ export const validateAlias = (alias: string) => {
   }
 
   return valid;
-};
\ No newline at end of file
+};
+
+export const getAliasRoute = (location: Location, data: Alias[], searchParams: URLSearchParams) => {
+  const currentURL = location.pathname.substring(1);
+  let redirectURL = '';
+  // we need to check if the whole url here is an alias, so we can redirect
+  const foundAlias = data.filter((d) => d.alias === currentURL && d.enabled)[0];
+  if (foundAlias) {
+    redirectURL = generateURLFromAlias(foundAlias);
+    return redirectURL;
+  } else {
+    for (const d of data) {
+      const aliasOnPage = searchParams.get('alias');
+      if (aliasOnPage) {
+        // if the alias fits the alias on this page, but the URL is diferent, we redirect user to the new URL
+        // if we have the same alias and its enabled and its not empty
+        if (d.alias !== '' && d.enabled && d.alias === aliasOnPage) {
+          const newAliasPath = resolvePath(d.pathAndParams);
+          const urlParams = new URLSearchParams(newAliasPath.search);
+          urlParams.set('alias', d.alias);
+          // we confirm either the url parameters does not match or the url path doesnt
+          if (!isEqual(urlParams, searchParams) || newAliasPath.pathname !== location.pathname) {
+            // we then redirect to the alias route, since the view listening to this alias has an outdated URL
+            redirectURL = `${newAliasPath.pathname}?${urlParams.toString()}`;
```

Comment:
I think you might not have to call the toString in a template string?
I might be wrong...

Reply:
you're right, I've updated it